src/lobster/mcp/tools/representations.py

```diff
@@ -0,0 +1,125 @@
+"""Sequence representation tools for Lobster MCP server using FastMCP best practices."""
+
+import logging
+from dataclasses import dataclass
+from typing import Literal
+
+import torch
+
+from .tool_utils import _load_model
+
+logger = logging.getLogger("lobster-fastmcp-server")
+
+
+@dataclass
+class SequenceRepresentationResult:
+    """Result of sequence representation extraction."""
+
+    embeddings: list[list[float]] | list[list[list[float]]]
+    embedding_dimension: int
+    num_sequences: int
+    representation_type: str
+    model_used: str
+
+
+def get_sequence_representations(
+    model_name: str,
+    sequences: list[str],
+    model_type: Literal["masked_lm", "concept_bottleneck"],
+    representation_type: Literal["cls", "pooled", "full"] = "pooled",
+) -> SequenceRepresentationResult:
```

Comment:
Have you tested whether the dataclass output works? The only thing I can find in the docs on more complex data is Pydantic models: https://gofastmcp.com/servers/tools#pydantic-models